### ISSUE TITLE
Write  Use Cases for the uploader tool

### ISSUE THREAD

**Issue by jellegerbrandy** [Paratii-Video/paratii-player#261]
We need to ciscover what we expect that users can do with our upload tool before we can actually implement it :-)

Type of experience we are thinking of "K.im demo" and "Giphy uploader"

Possible use cases we coudl write:

- Casual user Maria uploads a single movie from his phone
- Youtuber Toddy migrates his YT channel to paratii
- Publisher nytimes publishes 20 videos daily in different channels
- etc etc.

The idea is to first map a number of user experiences, reach consensus between us about what that expereience should look like. Once we have that, we can start designing screens, etc.... 

Ref for use cases: https://en.wikipedia.org/wiki/Use_case

**Comment by felipegaucho** [comment]
- **1. Casual user Maria uploads a single movie from her phone.**
Trigger: sees an ad somewhere, or gets a direct recommendation, and wants an easy way to try Paratii.
Basic Flow: navigates to uploading URL, finds her way, fetches file from her phone media library, uploads it on a shitty connection, takes some time (here we can present useful info as text too @jellegerbrandy) to finish this, gets a URL for her video. She can then be presented with further options: create a playlist, for example; facebook connect and see what friends of her have posted on paratii?; etc. Specially on mobile, I see this proces as being punctuaded by little helpers (think using dropbox or Slack for the first time).

- **2. Bob sees his friend doing a triple backflip. Wants to tease his friends, posting a tweet with the video and tagging 10 friends + asking them to pay U$1, if they dare, to watch this mundane miracle. He asks the guy to do it again so he can record it on his mobile**
Trigger: sees something spectacular, knows he can share and monetise this easily on Paratii.
Basic flow: wants to record something from camera and upload it, or even record from camera directly into the uploader. 

- **3. I'm watching a conference where a friend of mine is going to speak next. I want to film a part of it, upload and share, and incentivise people to share it further**
Trigger: sees something spectacular, knows he can share and potentialise this on Paratii.
Basic flow: similar to above, with emphasis on monetisation / revenue sharing at the end.

- **4. Youtuber Toddy finds out about Paratii and decides to migrate or mirror his YT channel here**.
Trigger: read about, heard from fellow producers, seeking monetisation. Saw a video by a friend and remembered it, so he clicks on "upload here" somewhere alongside the video or within the player, and goes to the upload tool.
Basic Flow: searches for "mirror Youtube"-like option, expects to paste link from his youtube channel, then get a selection/read of all his videos, option/checkboxes to choose the ones he want + filter to select "non-monetized only", or stuff like this, then clicks on "go", an waits till all those are mirrored. Then can "replicate" playlists (hmm, sounds complicated, can we get all these metadata?), or "create playlists" with the videos uploaded. Also can have a view of many (a list, before it becomes a dashboard), and each one's monetisation option.

- **5. My dad wants to save all our family archives he found in VHS (heavy transcoding involved :) ) into an uncensorable, resistant system, in a way that's easily watchable by my aunt, my grandma and my cousin**.
Trigger: Crazy adult people who like technology sometimes have this strange impulses.
Basic flow: Searches for paratii on google, gets to uploader tool, sees he can upload mp4 from his machine, convert VHS to mp4 on his machine, upload batch of files, create playlist sorting files by events, share playlist with family by email or on family's whatsapp group.

- **6. Big publisher wants to move the "tech" section videos to Paratii. IT or media guy there has to move part of the catalogue through our uploader. They have all files and metadata in their closed system.**
Trigger: economic interest, reads about the player or has a commercial contact, directs an employee to look over this.
Basic flow: fetches all the data miraculously intelligently (or mannualy), mirrors the metadata on Paratii (miracle 2), or uploads a batch of files and a .csv with metadata?, creates playlists with the videos if needed, overlays a watermark (??), maybe define somethings in regard to recommendations (e.g. my embed will only bring recommendations from this accredited producers, and then checkbox a few suggestions or set own ones), gets custom embed code. Probably verify domain (https://faq.dailymotion.com/hc/en-us/articles/115008958347-Verify-your-domains) at some point (for joining early ad partners program, maybe?).

- **7. Catarse, a crowdfunding platform, wants to allow their users to host project's videos on Paratii. In the "video" step of creating a project, you upload a file or paste a youtube link, I think. I am a user who wants to paste a Paratii link.**
>> What would take for a Paratii link to directly be "renderable" here, besides getting oin touch with the guys behind the service and getting them to allow it? Something special, technically?

- **8. I want to scrap all the local TV programs that are censored by my government and end up just being broadcast p2p, and upload them automatically**. 
Trigger: looking for uncensorability, a bit tech-hard, won't care much for non-friendliness.
Basic flow: I want to scrap sources of streams I define and automatically upload them. Can there be any API to facilitate this? Very extreme case =)

- **2 references I like a lot:**
[Giphy's upload](https://giphy.com/upload#help)
[K.im demo](https://www.youtube.com/watch?v=FSQOLT_uq7M)

- **6 basic options I think of**:
Upload from desktop/media collection; 
Record from webcam/mobile camera; 
Bulk upload; 
Import from URL (specific video); 
Mirror whole YT/Vimeo channel; 
Mirror playlist.

**Comment by pedrocasa** [comment]
gonna try my first one...

( in different point o view of cases from above)

9. Mike heard from his friends that his favourite youtube channel migrated to paratii and wants to see it in the new platform but couldn't find the link for it

Trigger: Heard about an amazing channel but does not know how to find it

steps

- Go to Paratii URL and uses discovery/search feature to search for the name of your favourite channel
- Find it 
- [ Perhaps necessary ] create an account to follow the channel and receives notifications about it on your email
- Give your feedback for the channel about its new videos

**Comment by jellegerbrandy** [comment]
Exactly. 

These should probably two use cases (find a channel, and another: subscribe to a channel), and it j needs more precision (go to paratii url, click on "discover", search for "John 
Smith" (= the name of your friend), see the result page (which as 24 results, among which videos created by yr friend, etc), click on the "channel" link, go the page channel page.

And another one for the subscription (which is perhaps out of scope for the upload tool?) 

Can you add this one to the wiki? https://github.com/Paratii-Video/paratii-player/wiki/Video-Upload-Use-Cases?

**Comment by pedrocasa** [comment]
Sure Thing, I uploaded here because I was not 100% secure about the information that I writed

**Comment by jellegerbrandy** [comment]
Okay, i moved some of these first scenarios here: https://github.com/Paratii-Video/paratii-usecases/issues
Closing this, let's continue in the new repo
